apps/server/src/form-instances/form-instances.service.ts

```diff
@@ -300,6 +300,42 @@ export class FormInstancesService {
       ...updatedSignature,
     };
 
-    return formInstance;
+    const allSigned = formInstance.signatures.every((sig) => sig.signed);
+
+    if (allSigned) {
+      await this.prisma.formInstance.update({
+        where: { id: formInstanceId },
+        data: { completed: true, completedAt: new Date() },
+      });
+    }
+
+    const updatedFormInstance = await this.findOne(formInstanceId);
+
+    if (!updatedFormInstance) {
+      throw new NotFoundException(
+        FormInstanceErrorMessage.FORM_INSTANCE_NOT_FOUND,
+      );
+    }
+
+    return updatedFormInstance;
+  }
+
+  async markFormInstanceAsCompleted(formInstanceId: string) {
+    const formInstance = await this.prisma.formInstance.findUnique({
+      where: { id: formInstanceId },
+    });
+
+    if (!formInstance) {
+      throw new NotFoundException(
+        FormInstanceErrorMessage.FORM_INSTANCE_NOT_FOUND,
+      );
+    }
+
+    const updatedFormInstance = await this.prisma.formInstance.update({
+      where: { id: formInstanceId },
+      data: { markedCompleted: true, markedCompletedAt: new Date() },
+    });
+
+    return updatedFormInstance;
```

Comment:
let's also make a ticket for adding tests for these new conditions and methods? i'd link the ticket onto this comment for visibility, and leave the ticket under backend backlog

Reply:
https://trello.com/c/c8i6zWCj/116-new-form-instances-tests